crates/matrix-sdk/src/client/mod.rs

```diff
@@ -3218,6 +3074,73 @@ impl Client {
     pub(crate) fn thread_subscription_catchup(&self) -> &ThreadSubscriptionCatchup {
         self.inner.thread_subscription_catchup.get().unwrap()
     }
+
+    /// Perform database optimizations if any are available, i.e. vacuuming in
+    /// SQLite.
+    ///
+    /// **Warning:** this was added to check if SQLite fragmentation was the
+    /// source of performance issues, **DO NOT use in production**.
+    #[doc(hidden)]
+    pub async fn optimize_stores(&self) -> Result<()> {
+        trace!("Optimizing state store...");
+        self.state_store().optimize().await?;
+
+        trace!("Optimizing event cache store...");
+        self.event_cache_store().lock().await?.as_clean().unwrap().optimize().await?;
```

Comment:
Won't this panic in case the event cache lock is dirty and we try to optimize the store?